### DISCUSSION TITLE
How to minimize mini buffer before running command?

### DISCUSSION THREAD

**Discussion by bradhowes** [minad/vertico#432] [Q&A]
I have an interactive Emacs lisp method that reads a function name -- `(interactive "aFunction: ")` -- which then uses the function to do some work on all of the lines of the current buffer. Here's the function:

```
(defun my/c++-edit-all-files-named-in-buffer (p)
  "Edit all files that are referenced in current buffer.
They are assumed to be relative to the buffer that holds
them. This routine moves to the first line and acts on each
line, calling the given P function on each one in turn.
"
  (interactive "aFunction: ")
  (let ((cwd default-directory))
    (point-min)
    (while (looking-at "^.+$")
      (let* ((partial (match-string-no-properties 0))
             (path (file-name-concat cwd partial))
             (buf (find-file-noselect path)))
        (message "Editing %s" partial)
        (with-current-buffer buf
          (funcall p))
        (kill-buffer buf))
      (beginning-of-line 2)))
  (message "Done."))
```

That all works just fine, but when it runs, the completion buffer stays open while my function runs. Is there a way to programmatically resize/clear/dismiss the completion window after satisfying the completion read operation? 

Although my question may seem to be about Emacs minibuffer completion, I'm using Vertico to have the nice view of functions and the powerful completion tools it provides, so I'm wondering if there is Vertico functionality to achieve this that I have missed.

**Comment by minad** [comment]
After completion, the minibuffer should automatically shrink (after `completing-read` returned). If you want to resize the minibuffer during Vertico operation, you may want to take a look at https://github.com/minad/vertico/wiki#automatically-shrink-vertico-for-embark-live or https://github.com/minad/vertico/wiki#adjust-number-of-visible-candidates-when-buffer-is-resized and experiment with the solutions there.
